pkg/arvo/app/chat.hoon

```diff
@@ -340,201 +222,197 @@
   ?+  i.wir
     (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
   ::
-  ::  %circles
       %circles
-    ?>  ?=(%circles -.rum)
-      =/  cis
+    (handle-rumor-circles rum)
+  ::
+      %circle
+    (handle-rumor-circle wir rum)
+  ::
+  ==
+::
+::  +handle-rumor-circles
+::
+++  handle-rumor-circles
+  |=  rum=rumor:hall
+  ^-  (quip move _this)
+  ?>  ?=(%circles -.rum)
+  =/  cis
+    ?:  add.rum
+      (~(put in circles.str) cir.rum)
+    (~(del in circles.str) cir.rum)
+  =.  str
+    %=  str
+      circles  cis
+      peers
         ?:  add.rum
-          (~(put in circles.str.sta) cir.rum)
-        (~(del in circles.str.sta) cir.rum)
-      =/  str
-        %=  str.sta
-          circles  cis
-          peers
-            ?:  add.rum
-              (~(put by peers.str.sta) [our.bol cir.rum] ~)
-            (~(del by peers.str.sta) [our.bol cir.rum])
-        ==
-      :-  (send-chat-update [[%circles cis] str])
-      this(str.sta str)
+          (~(put by peers.str) [our.bol cir.rum] ~)
+        (~(del by peers.str) [our.bol cir.rum])
+    ==
+  :_  this(str str)
+  (send-chat-update [[%circles cis] str])
+::
+++  handle-rumor-circle
+  |=  [wir=wire rum=rumor:hall]
+  ^-  (quip move _this)
+  ?>  ?=(%circle -.rum)
+  ?+  -.rum.rum
+    [~ this]
   ::
+  ::  %gram: new message
   ::
-  ::  %circle: fill remote configs with message data
+      %gram
+    (handle-rumor-circle-gram wir rum.rum)
   ::
-      %circle
-    ?>  ?=(%circle -.rum)
-    =*  sto  rum.rum
-    ?+  -.sto
+  ::  status: status update
+  ::
+      %status
+    (handle-rumor-circle-status rum.rum)
+  ::
+  ::  %config: config has changed
+  ::
+      %config
+    =*  circ  cir.rum.rum
+    ::
+    ?+  -.dif.rum.rum
```

Comment:
I'd split the different `%config` out into their own arms also. The ones above looks so much better now!

Reply:
Done! Thanks.